src/plugin/svipc/sysvipcwrappers.cpp

```diff
@@ -183,9 +188,72 @@ shmdt(const void *shmaddr)
 // in order to install its own hooks. For us, shmdt() is the only interesting
 // one. Instead of giving the address of our wrapper to the hook library, we
 // want to return the address in libc. See PR #472 for details.
+static int dlsym_addr_instance = 0;
+static const char *dlsym_symbol = NULL;
+static void *dlsym_retval;
+// extern "C"
+static int
+callback(struct dl_phdr_info *info, size_t size, void *data) {
+  void *handle = dlopen(info->dlpi_name, RTLD_LAZY);
+  dlerror(); // initialize for errors
+  void *address = dlsym(handle, dlsym_symbol);
+  if (address == NULL && dlerror() != NULL) {
+    dlsym_retval = dlsym(handle, dlsym_symbol); // Set dlerror to error string
```

Comment:
This seems redundant. Wouldn't dlerror already be set to the right value because of the previous call?

Reply:
I've modified the comment for clarity:
>  // Clear any old errors; will return non-null only on new error

Reply:
I think you misread my comment. It was about the second dlsym() call on line 201.